### DISCUSSION TITLE
Suggestion - add list of writeable registers to readme

### DISCUSSION THREAD

**Discussion by lamixer** [nathanmarlor/foxess_modbus#497] [Support] [answered]
Hello!

I wanted to set up a HA automation to set min soc on-grid to a high value when Octopus 'Intelligent Dispatch' goes ON . In your readme you have an example of how to "write a service" but I did not know to what 'Start Address' I wanted to write.

I dug for it and was able to find the info here: (https://github.com/nathanmarlor/foxess_modbus/blob/main/custom_components/foxess_modbus/entities/entity_descriptions.py) so I did get it to work!

To help users I think it would be great if you could keep a list of registers that the integration knows how to write (though I suppose one could use it to write to custom / unsupported registers also).

For example:

To set "min_soc_on_grid" write to address 41011 a numeric value from 0 to 100.

If this list exists somewhere here already, sorry I missed it! I know there are docs floating around the internet that have a list, but would be great for you to keep one in an easy-to-find place that shows what you have implemented.

I see that 'Update Charge Periods" service has more hand-holding. So the ultimate would be to have specific services like that to write each register that shows supported values (numeric range or choice of options in a menu like for work mode).

**Comment by canton7** [comment, marked as answer]
We expose entities which let you write to common values, such as min soc. See `number.min_soc_on_grid`. Set that using a service (such as `number.set_value`) rather than fiddling about with registers.

Setting registers directly is really an advanced use-case, and we should not be encouraging people to do it.

**Reply by lamixer** [reply]
Oh, so I'm doing it wrong! I guess I need to learn how to use Home Assistant better. How to use an automation to set a value.

**Reply by canton7** [reply]
```yaml
service: number.set_value
target:
  entity_id: number.min_soc_on_grid
data:
  value: 30
```

**Reply by lamixer** [reply]
Thank you! I did it with the GUI editor using the correct method you pointed out to me and this is the yaml it produced (same effect I'm sure)!

```
service: number.set_value
metadata: {}
data:
  value: "45"
target:
  entity_id: number.foxess_min_soc_on_grid
```
